src/main/jbake/content/blog/2016/10/18/hawkular-metrics-openshift-and-grafana.adoc

```diff
@@ -0,0 +1,210 @@
+
+But still, this is probably not what we would expect. What happens here is that the _$app_ variable we set up in queries is resolved into as many _container_name_ as there are, and result in the same number of series in a single graph. We can change that behaviour very easily thanks to a nice feature of Grafana: on the existing row, to the left, open the Row editor and in _Templating options_ activate duplication from variable _app_. Save and refresh the browser.
+
+ifndef::env-github[]
+image::/img/blog/2016/2016-10-17-grafana-apps-rows.png[Applications per row]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/2016-10-17-grafana-apps-rows.png[Applications per row]
+endif::[]
+
+That's better! By turning on row duplication based on our variable, Grafana has created 3 rows, and for each one it provides to the Hawkular plugin just one value of _$app_ at a time.
+
+Now we can monitor our microservices quite easily! If we don't want to see the _docker-build_ instance, just filter it out with the top combo.
+
+We will just add a little enhancement to the dashboard, to make easier to understand which row is for which app. On the first row, add a new _Text_ panel, set its title empty, span 2, height 100px, mode HTML and content:
+```
+<center><p style='font-size: 40pt'>$app</p></center>
+```
+
+It will display the microservice name. After some layout arrangement, here is the final result:
+
+ifndef::env-github[]
+image::/img/blog/2016/2016-10-17-grafana-apps-rows-2.png[Applications per row bis]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/2016-10-17-grafana-apps-rows-2.png[Applications per row bis]
+endif::[]
+
+You can play around with OpenShift. Scale some pods up and down, and you'll get the metrics updated in Grafana. Just note that on downscaling, you'll have to wait a little bit (5 minutes) before seeing the _Live_ metrics on the _Singlestat_ panel being updated. This is because we're not sure if the absence of data is due to a pod being shut down, or a simple delay between measurements.
+
+=== It's Just a Beginning
+
+Thanks to the metrics provided in OpenShift, you can build more elaborate dashboards. Just change the _desciptor_name_ tag and see what's interesting for you, there are metrics on memory, CPU, network and filesystem.
+
+But that's just the starting kit! link:http://www.hawkular.org/hawkular-clients/[The Hawkular Metrics ecosystem] is rich and keeps growing, including a Wildfly agent, a Vert.x plugin, a DropWizard reporter, etc. And if that's not enough for you, it's very easy to integrate your own metrics: either through the client libraries or by directly calling the link:http://www.hawkular.org/docs/rest/rest-metrics.html[Metrics REST API].
+
```

Comment:
I try but can't find a good formulation, as I'm mostly talking about the metrics API consumers here, Alerts integration seems a bit different, no?